src/controllers/chess-game-controller.js

```diff
@@ -52,26 +60,108 @@ export default class ChessGameController {
     });
 
     this.eventBus.addEventListener("piece-move", (event) => {
-      const chessPieceElement = this.board.chessPieces.find(
-        (element) => element.piece.id === event.detail.pieceId
-      );
+      this.pieceMoveHelper(event)
+    });
+  }
 
-      const destinationSquare = event.detail.to;
-      const destinationSquareRank = destinationSquare[0];
+  pieceMoveHelper(event) {
+    const chessPieceElement = this.board.chessPieces.find(
+      (element) => element.piece.id === event.detail.pieceId
+    );
 
-      chessPieceElement.remove();
-      const destination = this.board.getSquare(...destinationSquare);
-      destination.appendChild(chessPieceElement);
+    const destinationSquare = event.detail.to;
+    const destinationSquareRank = destinationSquare[0];
 
-      if (
-        chessPieceElement.piece.name === "Pawn" &&
-        (destinationSquareRank === 7 || destinationSquareRank === 0)
-      ) {
-        this.game.board.willRotate = false;
-        displayPromotionDialog(this.promotionDialog);
-      }
+    chessPieceElement.remove();
+    const destination = this.board.getSquare(...destinationSquare);
+    destination.appendChild(chessPieceElement);
+
+    if (
+      chessPieceElement.piece.name === "Pawn" &&
+      (destinationSquareRank === 7 || destinationSquareRank === 0)
+    ) {
+      this.game.board.willRotate = false;
+      displayPromotionDialog(this.promotionDialog);
+    }
+  }
+
+  handleMove() {
+    const activePlayer = this.game.getActivePlayer();
+
+    const color = activePlayer.color;
+    if (activePlayer.opponent.moves.length === 0) {
+      setTimeout(() => {
+        if (this.game.isPlayerInCheck()) {
+          const checkmate = `Checkmate! ${color} player wins!`;
+          declareWinner(
+            color,
+            this.game,
+            this.turn,
+            this.endGameDialog,
+            checkmate
+          );
+        } else {
+          const stalemate = "Stalemate";
+          declareDraw(this.game, this.endGameDialog, stalemate);
+        }
+      }, 500);
+      this.game.board.willRotate = false;
+    }
+
+    if (this.game.board.willRotate === true) {
```

Comment:
`=== true` is almost always redundant
```suggestion
    if (this.game.board.willRotate) {
```

Reply:
Given the goal of coalescing our handlers, you can add this to the end of the new `handlePieceMove` (see my other comment)